Add a release decision matrix to the release docs (#67084)
woocommerce/woocommerce: docs/contribution/releases/decision-matrix.md

---
post_title: Release Decision Matrix
sidebar_label: Decision Matrix

> sidebar_position: 5

Reviewer: Isn't this already used by the prereleases doc? I imagine it will be fine (order should be alphabetical I believe)

Author: It is - and monitoring and troubleshooting already share 7 in this folder, with the alphabetical tie-break putting them in the intended order. Same here: decision-matrix sorts before prereleases, which lands it right after readiness, where it belongs. Left as is.

---

# Release Decision Matrix

Every finding against a release gets a verdict - at the readiness review, at the go/no-go, and when weighing whether an already-shipped issue warrants a point release. This page is the criteria for picking that verdict. The decision points themselves are described in [Readiness and Go/No-Go](/docs/contribution/releases/readiness); the criteria apply to Core and Woo-owned extension releases, for all release types.

> The matrix guides the people making the call - the release lead and the Product DRI. It gives no automated check blocking power.

Author: Reworded in 102028001d: "It gives automated checks no power to block a release."

> The matrix guides the people making the call - the release lead and the Product DRI. It gives automated checks no power to block a release.

Reviewer: Technically, CI checks do block merges. Maybe this is clearer:

```suggestion
The matrix guides the people making the call - the release lead and the Product DRI. It does not change which automated checks gate a build.
```

Author: Fixed in 5e2a12f: applied your suggestion.

## Verdicts

| Verdict | Meaning |
| --- | --- |
| **Release-blocking** | The release does not ship with this issue in it: hold the date or remove the change. Found after shipping: point release now. |
| **Fix in a point release** | Ship on schedule. The fix has an issue, an owner, and a target release *before* the go decision - without those, it re-enters as release-blocking. |
| **Next release** | Ship. The fix rides the next scheduled release. |
| **Not a bug** | Expected behavior, or an invalid finding. |

> A finding that cannot be reproduced on two independent stacks does not block the release - the issue stays open with a named owner rather than being closed as not a bug.

Author: Good catch - the wording included a one-stack repro, which the floor forbids. Reworded in 102028001d: a finding that reproduces anywhere keeps its impact-class verdict; the two-stack rule only governs findings nobody can reproduce.

> A finding no one can reproduce - tried on at least two independent stacks - does not block the release; the issue stays open with a named owner rather than being closed as not a bug. A finding that reproduces anywhere keeps its impact-class verdict.

Reviewer: The RC checklist in this PR says every open finding must get one of the four verdicts. But a finding that nobody can reproduce gets none of them: it does not block the release, and this paragraph says we should not close it as "not a bug" either. Maybe "Next release, with a named owner"?

Author: Fixed in 5e2a12f: an unreproducible finding now gets "Next release, with a named owner", so every open finding maps to one of the four verdicts.

## Default verdict by impact class

| Impact class | Default verdict | Example |
| --- | --- | --- |
| Checkout or revenue interruption - a store cannot take payments | Release-blocking; no modifiers apply | - |

> | Site down or fatal on load, in Core or any Woo-owned extension - including older extension versions running with the new Core | Release-blocking | [#64394](https://github.com/woocommerce/woocommerce/pull/64394) forced the 10.9.0 revert; [#65957](https://github.com/woocommerce/woocommerce/pull/65957) rescheduled 11.0.0 after a fatal surfaced on an early deployment |

Reviewer: "10.9.0 revert" could be misunderstood that the release itself was reverted. Maybe:

```suggestion
| Site down or fatal on load, in Core or any Woo-owned extension - including older extension versions running with the new Core | Release-blocking | [#64394](https://github.com/woocommerce/woocommerce/pull/64394) forced the revert after 10.9.0; [#65957](https://github.com/woocommerce/woocommerce/pull/65957) rescheduled 11.0.0 after a fatal surfaced on an early deployment |
```

Author: Fixed in 5e2a12f: applied your suggestion.

| Data loss or corruption | Release-blocking | - |
| A break in a public contract or shared surface with consumers in the wild - PHP API, hooks, REST, datastore behavior | Release-blocking | [#65595](https://github.com/woocommerce/woocommerce/pull/65595) changed datastore behavior extensions relied on and forced a revert; [#66382](https://github.com/woocommerce/woocommerce/pull/66382) removed released public methods, restored within a day by [#66822](https://github.com/woocommerce/woocommerce/pull/66822) |

> | A contract change whose surface has no known consumers, or that went through the deprecation path in the repository's `AGENTS.md` | Fix in a point release, or next release | - |

Reviewer: Doesn't that conflict with `point-releases.md` where we say that point releases can't contain breaking changes?

Author: Fixed in 5e2a12f: the row's default is now "Next release", so the matrix no longer routes a contract change toward a point release.

> | A contract change whose surface has no known consumers, or that went through the deprecation path in the repository's `AGENTS.md` | Fix in a point release, or next release | - |

Reviewer: BTW this page is published on developer.woocommerce.com, where readers do not have the repository files, and AGENTS.md is not a link here. Should we link directly?

Author: Fixed in 5e2a12f: linked to the file on trunk. It is the repo-root AGENTS.md - the deprecation path lives in its backward-compatibility section.

| WordPress forward-compatibility - the release breaks under a WordPress version shipping before our next release | Fix in a point release scheduled before the WordPress date; release-blocking if the breakage is site-down class | [#67061](https://github.com/woocommerce/woocommerce/pull/67061) shipped WordPress 7.1 fixes inside the 11.0.0 rebuild, because 7.1 lands before 11.1.0 |
| Broken upgrade path - the update itself fails or leaves a site inconsistent | Release-blocking if deterministic; fix in a point release if transient and self-healing | The 10.9.1 upgrade race self-healed on cache refresh and produced a single report - no point release was needed |
| A regression in a default-on, merchant-facing feature | Release-blocking without a workaround; fix in a point release with one | - |

> | A regression behind a flag that is off by default | Fix in a point release, or next release | An 11.0 REST API field inconsistency was judged non-blocking because the flag is off by default, with the rename scheduled for a later release |

Reviewer: This row and the "Feature flags, by trajectory" modifier on line 46 cover the same case. If a reader applies both, the verdict drops two levels instead of one. Intentional?

Author: Not intentional - fixed in 5e2a12f: a verdict moves at most one level in total, and a modifier does not reapply when the matched row already prices it in. The flag row's default is also "Next release" now, which removes the stacking case in practice.

- **Blast radius.** How many stores are exposed, and how fast. Managed hosting platforms can deploy an update to every site at once rather than following WordPress.org's staged rollout, so managed-fleet exposure weighs heavier than install counts alone. This extends the scope and commonality criteria in [Point Releases](/docs/contribution/releases/point-releases).
- **Workaround.** A documented workaround that merchants can apply themselves demotes one level - never past the floor. This matches the workaround criterion in [Point Releases](/docs/contribution/releases/point-releases) and [Release Monitoring](/docs/contribution/releases/monitoring).

## Response times

| Verdict | Response time |
| --- | --- |
| Release-blocking | The verdict is recorded the same day the finding is raised; the release does not proceed to the next run-book step until it is. |
| Fix in a point release | The fix is scheduled - issue, owner, target release - before the go decision. For findings that surface after shipping: within the [monitoring window](/docs/contribution/releases/monitoring) (3 days after a major release, 1 day after a point release). |

> | Next release | The issue is filed and milestoned before the go decision. |

Reviewer: "Not a bug" has no row. And the "Next release" row says "before the go decision", which does not work for a finding that appears after the release has already shipped.

Author: Fixed in 5e2a12f: added a "Not a bug" row, and the "Next release" row now covers findings that surface after shipping via the monitoring window.

These are verdict-and-scheduling times, not fix-delivery guarantees.

## Decision rules

1. **Revert over rush.** When a fix cannot get the same validation the original change had before the release date, remove the change instead of patching under pressure.
2. **Silence is not a pass.** Every signal read at the readiness review and go/no-go has a named responder. An unreviewed signal counts as missing, not green.

> 3. **Verdicts are recorded, with names.** The release lead and the Product DRI record each verdict on the linked issue. Classification disputes end there.

Reviewer: "Classification disputes end there" can be read as "disputes are stopped". I think you mean that the discussion happens on the linked issue and is written down there?

Author: Right - fixed in 5e2a12f: disputes are argued and settled on the linked issue, so the reasoning is written down where the verdict lives.
